application/models/Qsl_model.php

```diff
@@ -44,41 +46,41 @@ function saveQsl($qsoid, $filename) {
 
         $data = array(
             'qsoid' => $clean_id,
-            'filename' => $filename
+            'file_id' => $clean_file_id
         );
 
         $this->db->insert('qsl_images', $data);
 
         return $this->db->insert_id();
     }
 
-    function deleteQsl($id) {
+    function deleteQsl($qsl_id) {
         // Clean ID
-        $clean_id = $this->security->xss_clean($id);
+        $clean_id = $this->security->xss_clean($qsl_id);
 
         // be sure that QSO belongs to user
         $CI =& get_instance();
         $CI->load->model('logbook_model');
         if (!$CI->logbook_model->check_qso_is_accessible($clean_id)) {
-            return;
+            //return;
```

Comment:
Why did you comment out this? This must be there to check that the user is allowed to delete the qsl. If no rights, return.

Reply:
Here should pass $qso_id instead of $qsl_id, the qso accessible check here may block some of request unexpectedly.

Reply:
I see, that's what you meant with needing another patch?

Reply:
Yes, a new query may needed here.